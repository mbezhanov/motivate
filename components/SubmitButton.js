import React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import { Octicons } from '@expo/vector-icons';
import Colors from '../constants/Colors';
import Sizes from '../constants/Sizes';
import PropTypes from 'prop-types';

const submitButton = (props) => (
  <TouchableOpacity style={styles.button} onPress={props.onPress}>
    <Octicons name="check" size={Sizes.L} color={Colors.HIGHLIGHT} />
  </TouchableOpacity>
);

submitButton.propTypes = {
  onPress: PropTypes.func
};

const styles = StyleSheet.create({
  button: {
    marginRight: Sizes.M
  }
});

export default submitButton;
