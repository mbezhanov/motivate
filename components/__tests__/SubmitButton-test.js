import React from 'react';
import renderer from 'react-test-renderer';
import SubmitButton from '../SubmitButton';

jest.mock('react-native/Libraries/Components/Touchable/TouchableOpacity', () => 'TouchableOpacity');

jest.mock('@expo/vector-icons', () => {
  const React = require('react');
  class Octicons extends React.Component {
    render() {
      return React.createElement('Octicons', this.props, this.props.children)
    }
  }
  return { Octicons };
});

describe('SubmitButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SubmitButton />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('can use a custom onPress handler', () => {
    const stub = () => '123';
    const tree = renderer.create(<SubmitButton onPress={stub} />).toJSON();
    expect(tree.props.onPress).toEqual(stub);
  });
});
