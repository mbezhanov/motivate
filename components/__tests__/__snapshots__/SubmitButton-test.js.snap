// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton renders correctly 1`] = `
<TouchableOpacity
  style={
    Object {
      "marginRight": 16,
    }
  }
>
  <Octicons
    color="#fff"
    name="check"
    size={21}
  />
</TouchableOpacity>
`;
